### PR TITLE
feat(fgs/quota): add new data source to query resource quotas

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Supports a new data source that used to query available resource quotas.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new data source to query resource quotas.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccDataSourceQuotas_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccDataSourceQuotas_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceQuotas_basic
=== PAUSE TestAccDataSourceQuotas_basic
=== CONT  TestAccDataSourceQuotas_basic
--- PASS: TestAccDataSourceQuotas_basic (21.29s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       21.333s
```

### CODE REVIEW

File: docs/data-sources/fgs_quotas.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  The [quotas](#quotas_struct) structure is documented below.
+
+<a name="quotas_struct"></a>
+The `quotas` block supports:
+
+* `type` - The resource type corresponding to quota.
+
+* `unit` - The unit of usage.
+  
+  -> For `fgs_func_code_size`, the unit is `MB`. In other scenarios, there is no unit.
```
Comment (deer-hang) [inline docs/data-sources/fgs_quotas.md]: It’s hard to read the meaning of `fgs_func_code_size` here. Please list and explain the `type` field enumeration values.

File: huaweicloud/services/acceptance/fgs/data_source_huaweicloud_fgs_quotas_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+func TestAccDataSourceQuotas_basic(t *testing.T) {
+	var (
+		all = "data.huaweicloud_fgs_quotas.test"
+		dc  = acceptance.InitDataSourceCheck(all)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckFgsAgency(t)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/fgs/data_source_huaweicloud_fgs_quotas_test.go]: It is recommended to add comments to the environment variables used.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccDataSourceQuotas_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccDataSourceQuotas_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceQuotas_basic
=== PAUSE TestAccDataSourceQuotas_basic
=== CONT  TestAccDataSourceQuotas_basic
--- PASS: TestAccDataSourceQuotas_basic (20.73s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       20.803s
```

File: huaweicloud/services/acceptance/fgs/data_source_huaweicloud_fgs_quotas_test.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/fgs/data_source_huaweicloud_fgs_quotas_test.go]: there is no need to use this env for the function create, in this scenario, the function does not needs the other service's privileges.

File: docs/data-sources/fgs_quotas.md (hunk 1 above)
Comment (Lance52259) [reply docs/data-sources/fgs_quotas.md]: done.

**Pull request merged**